File: app/src/services/staffService.js
import api from '../config/api';

export const staffService = {
  createStaff: async (formData) => {
    const response = await api.post('/staff', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data;
  },

  getMyStaff: async () => {
    const response = await api.get('/staff/my-staff');
    return response.data;
  },

  getSalonStaff: async (salonId) => {
    const response = await api.get(`/staff/salon/${salonId}`);
    return response.data;
  },

  updateStaff: async (staffId, formData) => {
    const response = await api.put(`/staff/${staffId}`, formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data;
  },

  deleteStaff: async (staffId) => {
    const response = await api.delete(`/staff/${staffId}`);
    return response.data;
  },
};

File: luxtril_flutter/lib/services/staff_service.dart
import 'api_client.dart';
import '../models/staff.dart';

class StaffService {
  final ApiClient _api = ApiClient();

  Future<List<Staff>> getSalonStaff(String salonId) async {
    final response = await _api.get('/staff/salon/$salonId');
    final List<dynamic> data = response['staff'] ?? response['data'] ?? [];
    return data.map((s) => Staff.fromJson(s)).toList();
  }

  Future<Staff> createStaff(Map<String, dynamic> data) async {
    final response = await _api.post('/staff', body: data);
    return Staff.fromJson(response['staff'] ?? response);
  }

  Future<Staff> updateStaff(String id, Map<String, dynamic> data) async {
    final response = await _api.put('/staff/$id', body: data);
    return Staff.fromJson(response['staff'] ?? response);
  }

  Future<void> deleteStaff(String id) async {
    await _api.delete('/staff/$id');
  }

  Future<Staff> toggleAvailability(String id) async {
    final response = await _api.put('/staff/$id/toggle-availability');
    return Staff.fromJson(response['staff'] ?? response);
  }
}
